File: update_bronco.py
import asyncio
from sqlalchemy.ext.asyncio import create_async_engine, AsyncSession
from sqlalchemy.orm import sessionmaker
from sqlalchemy import text
import os

DATABASE_URL = "postgresql+asyncpg://neondb_owner:npg_8qBxyj5dFpAL@ep-cool-butterfly-a2b1yqf3-pooler.eu-central-1.aws.neon.tech/neondb?ssl=require"

engine = create_async_engine(DATABASE_URL, echo=False)
AsyncSessionLocal = sessionmaker(engine, class_=AsyncSession, expire_on_commit=False)

async def main():
    async with AsyncSessionLocal() as session:
        query = text("""
            UPDATE "PruebasFisicas" p
            SET broncotest_20m = p.broncotest, broncotest = NULL
            FROM jugadores_propios j
            WHERE p.jugador_id = j.id
            AND p.fecha = '2025-11-01'
            AND p.broncotest IS NOT NULL
            AND (
                j.nombre ILIKE '%Adrian%' OR 
                j.nombre ILIKE '%Alex%' OR 
                j.nombre ILIKE '%Alexis%' OR 
                j.nombre ILIKE '%Xavi%' OR 
                j.nombre ILIKE '%Victor%'
            )
        """)
        
        result = await session.execute(query)
        await session.commit()
        print(f"Updated {result.rowcount} rows.")

if __name__ == "__main__":
    asyncio.run(main())

File: update_bronco.js
const { Client } = require('pg');

const client = new Client({
    connectionString: 'postgresql://neondb_owner:npg_U0vCalO7zufg@ep-bold-recipe-alp1x8ep-pooler.c-3.eu-central-1.aws.neon.tech/neondb?sslmode=require&channel_binding=require'
});

async function run() {
    try {
        await client.connect();

        // Find players matching criteria
        const playersQuery = `
      SELECT id, nombre, apellidos FROM jugadores_propios
      WHERE nombre ILIKE '%Adrian%'
         OR nombre ILIKE '%Alex%'
         OR nombre ILIKE '%Alexis%'
         OR nombre ILIKE '%Xavi%'
         OR nombre ILIKE '%Victor%'
    `;
        const playersRes = await client.query(playersQuery);
        const validIds = playersRes.rows.map(r => r.id);
        console.log(`Found ${validIds.length} players modifying Bronco test.`);

        if (validIds.length > 0) {
            // Update tests
            const updateQuery = `
        UPDATE "pruebas_fisicas"
        SET broncotest_20m = broncotest, broncotest = NULL
        WHERE fecha = '2025-11-01'
          AND broncotest IS NOT NULL
          AND jugador_id = ANY($1::uuid[])
      `;
            const updateRes = await client.query(updateQuery, [validIds]);
            console.log(`Updated ${updateRes.rowCount} physical tests.`);
        }

    } catch (err) {
        console.error('Error:', err);
    } finally {
        await client.end();
    }
}

run();
